### PR TITLE
Added ALDRYN_SEARCH_EXCLUDED_PLUGINS setting

### PR DESCRIPTION
**Pull request by DmitriySelischev**
This setting excludes specified CMS plugins from getting index data.
This setting is really helpful when we use plugins that are not compatible with aldryn_search such as: [djangocms-audio](https://github.com/divio/djangocms-audio), [djangocms-video](https://github.com/divio/djangocms-video).
This setting will also available for modules like djangocms-blog, that uses get_plugin_index_data from this aldryn_search addon.
Please, accept pull request

### CODE REVIEW

File: aldryn_search/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '0.4.1'
+__version__ = '0.4.2'
```
Comment (czpython) [inline aldryn_search/__init__.py]: version bumps are done per release. please exclude this from pr.

File: README.rst [hunk 2]
```diff
@@ -70,6 +70,13 @@ aldryn-search comes with an App Hook for django CMS, and a search view using Dja
 want to use this app hook, you can either subclass it and register it yourself, or set
 ``ALDRYN_SEARCH_REGISTER_APPHOOK`` to ``True``.
 
+If you wan't to exclude some cms plugins from indexing, you can specify ``ALDRYN_SEARCH_PLUGINS_EXCLUDE`` setting like so::
+
+    ALDRYN_SEARCH_PLUGINS_EXCLUDE = [
+        "plugin_app1.PluginName1",
```
Comment (czpython) [inline README.rst]: cms plugin types are unique already. please change this to 

```
ALDRYN_SEARCH_EXCLUDED_PLUGINS = ['PluginA', 'PluginB']
```

File: README.rst [hunk 3]
```diff
@@ -70,6 +70,13 @@ aldryn-search comes with an App Hook for django CMS, and a search view using Dja
 want to use this app hook, you can either subclass it and register it yourself, or set
 ``ALDRYN_SEARCH_REGISTER_APPHOOK`` to ``True``.
 
+If you wan't to exclude some cms plugins from indexing, you can specify ``ALDRYN_SEARCH_PLUGINS_EXCLUDE`` setting like so::
+
+    ALDRYN_SEARCH_PLUGINS_EXCLUDE = [
```
Comment (czpython) [inline README.rst]: please change setting name to `ALDRYN_SEARCH_EXCLUDED_PLUGINS`

File: aldryn_search/helpers.py [hunk 4]
```diff
@@ -35,12 +36,14 @@ def get_cleaned_bits(data):
 
 
 def get_plugin_index_data(base_plugin, request):
+    excluded_plugins = getattr(settings, 'ALDRYN_SEARCH_PLUGINS_EXCLUDE', [])
```
Comment (czpython) [inline aldryn_search/helpers.py]: please do this outside of the function like:

```
EXCLUDED_PLUGINS = getattr(settings, 'ALDRYN_SEARCH_PLUGINS_EXCLUDE', [])
```

File: aldryn_search/helpers.py [hunk 5]
```diff
@@ -35,12 +36,14 @@ def get_cleaned_bits(data):
...
     text_bits = []
 
     instance, plugin_type = base_plugin.get_plugin_instance()
 
-    if instance is None:
-        # this is an empty plugin
+    if instance is None or hasattr(instance, '_meta') and hasattr(instance._meta,
```
Comment (czpython) [inline aldryn_search/helpers.py]: change to `if instance is None or instance.plugin_type in EXCLUDED_PLUGINS`

Comment (DmitriySelischev) [reply aldryn_search/helpers.py]: I did it first time, but some more tests became failed with object "Options" or something like this

Comment (DmitriySelischev) [reply aldryn_search/helpers.py]: Ok, i'll change it

**Comment by DmitriySelischev** [comment]
Review changes applied

**Comment by czpython** [comment]
Thanks!

**Pull request merged**